File: AutoPowerModeSwitcher/DebugUI.vb
Public Class DebugUI
    Private Sub CheckBox24_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox24.CheckedChanged
        If CheckBox24.Checked = True Then
            UserPreferencesModule.OverrideWarningLabel.Visible = True
            UserInterface.TargetProcessBox.Enabled = False
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DO NOT let user override target process on main form = True")
        Else
            UserInterface.TargetProcessBox.Enabled = True
            UserPreferencesModule.OverrideWarningLabel.Visible = Not True
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DO NOT let user override target process on main form = False")
        End If
    End Sub

    Private Sub CheckBox13_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox13.CheckedChanged
        If CheckBox13.Checked = True Then
            UserPreferencesModule.UserTrackingCheckbox.Checked = True
            UserPreferencesModule.UserTrackingCheckbox.Enabled = False
            UserPreferencesModule.OverrideWarningLabel.Visible = True
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "Now FORCING user-tracking features")
        Else
            UserPreferencesModule.UserTrackingCheckbox.Enabled = Not False
            UserPreferencesModule.OverrideWarningLabel.Visible = Not True
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "No longer forcing user-tracking features")
        End If
    End Sub



    Private Sub CheckBox1_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox1.CheckedChanged
        If CheckBox1.Checked Then
            InstallerModule.Show()
        Else
            InstallerModule.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox2_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox2.CheckedChanged
        If CheckBox2.Checked = True Then
            BootManagerUXPort.Show()
        Else
            BootManagerUXPort.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox8_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox8.CheckedChanged
        If Label14.Text = "True" Then
            If CheckBox8.Checked = True Then
                UserInterface.Show()
            Else
                UserInterface.Dispose()

            End If
        Else

        End If

    End Sub

    Private Sub CheckBox7_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox7.CheckedChanged
        If Label14.Text = "True" Then
            If CheckBox7.Checked = True Then
                CoreForm.Show()
            Else
                CoreForm.Dispose()



            End If
        Else

        End If

    End Sub

    Private Sub CheckBox6_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox6.CheckedChanged
        If Label14.Text = "True" Then
            If CheckBox8.Checked = True Then
                UserPreferencesModule.Show()
            Else
                UserPreferencesModule.Dispose()
            End If
        Else
        End If

    End Sub

    Private Sub CheckBox5_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox5.CheckedChanged

        If CheckBox8.Checked = True Then
            AutoPowerSwitcher.Show()
        Else
            AutoPowerSwitcher.Dispose()


        End If


    End Sub

    Private Sub CheckBox12_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox12.CheckedChanged

        Select Case MsgBox("Are you sure want to reset all the settings?", MsgBoxStyle.YesNo, "CryptedBytes Debugging Interface")
            Case MsgBoxResult.Yes
                My.Settings.Reset()
                My.Settings.Save()

            Case MsgBoxResult.No

        End Select


        CheckBox12.Checked = False

    End Sub

    Private Sub CheckBox11_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox11.CheckedChanged
        CheckBox11.Checked = False

        Throw New Exception("You have thrown an exception")

    End Sub

    Private Sub Statistics_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DeveloperForm is loaded into memory.")
        If My.Settings.DisableCoreForm = True Then
            CheckBox14.Checked = True
        Else
            CheckBox14.Checked = Not True

        End If
        Panel1.Location = New Point(12, 291)
        If My.Settings.DisableLegacyServices = True Then
            CheckBox19.Checked = not True
        Else
            CheckBox19.Checked = Not False
        End If

        If My.Settings.GroupPolicyForceAutoPilot = True Then
            CheckBox17.Checked = true
        End If
        If My.Settings.DisableSizeChangedEventChecks = True Then
CheckBox18.Checked = True

        End If
        ' Panel2.Location = New Point(4, 29)
        Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DeveloperForm is intialized.")
    End Sub

    Private Sub CheckBox10_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox10.CheckedChanged
        If CheckBox10.Checked = True Then
            CheckBox24.Checked = True
        Else
            CheckBox24.Checked = False


        End If
    End Sub

    Private Sub CheckBox4_CheckedChanged_1(sender As Object, e As EventArgs) Handles CheckBox4.CheckedChanged
        If CheckBox4.Checked Then
            CheckBox13.Checked = True
        Else
            CheckBox13.Checked = False
        End If
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles GetOpenForms.Tick
        If Application.OpenForms().OfType(Of UserInterface).Any Then
            CheckBox8.Checked = True
        Else
            CheckBox8.Checked = Not True

        End If
        If Application.OpenForms().OfType(Of CoreForm).Any Then
            CheckBox7.Checked = True
        Else
            CheckBox7.Checked = Not True

        End If
        If Application.OpenForms().OfType(Of UserPreferencesModule).Any Then
            CheckBox6.Checked = True
        Else
            CheckBox6.Checked = Not True

            If Application.OpenForms().OfType(Of AutoPowerSwitcher).Any Then
                CheckBox6.Checked = True
            Else
                CheckBox6.Checked = Not True

            End If
        End If
        Label14.Text = "True"
        GetOpenForms.Stop()
    End Sub

    Private Sub CheckBox9_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox9.CheckedChanged
        If CheckBox9.Checked Then
            BootManager.Show()
        Else
            BootManager.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub LinkLabel1_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) Handles LinkLabel1.LinkClicked
        Panel2.Dispose()
    End Sub

    Private Sub CheckBox3_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox3.CheckedChanged
        If CheckBox3.Checked Then
            Debugging.Show()
        Else
            Debugging.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox16_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox16.CheckedChanged
        If CheckBox16.Checked Then
            MemoryLeakTestingLab.Show()
        Else
            MemoryLeakTestingLab.Dispose()
            GC.Collect()
        End If
    End Sub

    Private Sub CheckBox14_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox14.CheckedChanged
        If CheckBox14.Checked Then
            My.Settings.DisableCoreForm = True
        Else
            My.Settings.DisableCoreForm = False

        End If
        My.Settings.Save()

    End Sub

    Private Sub CheckBox17_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox17.CheckedChanged
        If CheckBox17.Checked Then
            My.Settings.GroupPolicyForceAutoPilot = True
            UserPreferencesModule.userdetect.Checked = True
            UserPreferencesModule.userdetect.Enabled = False
            UserPreferencesModule.usernodetect.Checked = False
            UserPreferencesModule.usernodetect.Enabled = False
        Else
            UserPreferencesModule.userdetect.Checked = False
            UserPreferencesModule.userdetect.Enabled = True
            UserPreferencesModule.usernodetect.Checked = False
            UserPreferencesModule.usernodetect.Enabled = True
            My.Settings.GroupPolicyForceAutoPilot = False

        End If

        My.Settings.Save()


    End Sub

    Private Sub LinkLabel2_LinkClicked(sender As Object, e As LinkLabelLinkClickedEventArgs) Handles LinkLabel2.LinkClicked
        Panel1.Dispose()
    End Sub

    Private Sub CheckBox19_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox19.CheckedChanged
        If CheckBox19.Checked = True Then
            My.Settings.DisableLegacyServices = False
            MsgBox("My.Settings.DisableLegacyServices = False", MsgBoxStyle.Exclamation, "Boolean Value Set")
        Else
            My.Settings.DisableLegacyServices = True
            MsgBox(" My.Settings.DisableLegacyServices = True", MsgBoxStyle.Exclamation, "Boolean Value Set")
        End If

    End Sub

    Private Sub RadioButton2_CheckedChanged(sender As Object, e As EventArgs)

    End Sub

    Private Sub CheckBox18_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox18.CheckedChanged
        If CheckBox18.Checked = True Then
                My.Settings.DisableSizeChangedEventChecks = True 
        Else
                My.Settings.DisableSizeChangedEventChecks = false
        End If
    
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) 
      

    End Sub

    Private Sub CheckBox20_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox20.CheckedChanged
        Environment.failfast(1)
    End Sub

    Private Sub CheckBox21_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox21.CheckedChanged
      
        If CheckBox21.Checked = True Then
               GC.Collect
     MsgBox("Succesful.")
              CheckBox21.Checked = False
            Else

        End If
     
    End Sub

    Private Sub CheckBox22_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox22.CheckedChanged
        If CheckBox22.Checked = True Then
              CoreForm.errorLoggingText =  DateTime.Now.ToString("dd.MM.yyyy HH:mm.ss") + ": " + "CRITICAL: User thrown an exception" 
Call CoreForm.errorLoggingSystem
            CheckBox22.Checked = False

        End If
    End Sub

    Private Sub CheckBox23_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox23.CheckedChanged
          MsgBox("Form1's exact size is: " & UserInterface.formheightlabel.Text & " , " & UserInterface.formwidthlabel.text & " and is copied to your clipboard.")
        My.Computer.Clipboard.SetText(UserInterface.formheightlabel.Text & " , " & UserInterface.formwidthlabel.text)
        CheckBox23.Checked = false
    End Sub
End Class

File: AutoPowerModeSwitcher/DebugUI.cs
using System;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace AutoPowerModeSwitcher
{
    public partial class DebugUI
    {
        public DebugUI()
        {
            InitializeComponent();
        }
        private void CheckBox24_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox24.Checked == true)
            {
                My.MyProject.Forms.UserPreferencesModule.OverrideWarningLabel.Visible = true;
                My.MyProject.Forms.UserInterface.TargetProcessBox.Enabled = false;
                Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DO NOT let user override target process on main form = True");
            }
            else
            {
                My.MyProject.Forms.UserInterface.TargetProcessBox.Enabled = true;
                My.MyProject.Forms.UserPreferencesModule.OverrideWarningLabel.Visible = !true;
                Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DO NOT let user override target process on main form = False");
            }
        }

        private void CheckBox13_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox13.Checked == true)
            {
                My.MyProject.Forms.UserPreferencesModule.UserTrackingCheckbox.Checked = true;
                My.MyProject.Forms.UserPreferencesModule.UserTrackingCheckbox.Enabled = false;
                My.MyProject.Forms.UserPreferencesModule.OverrideWarningLabel.Visible = true;
                Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "Now FORCING user-tracking features");
            }
            else
            {
                My.MyProject.Forms.UserPreferencesModule.UserTrackingCheckbox.Enabled = !false;
                My.MyProject.Forms.UserPreferencesModule.OverrideWarningLabel.Visible = !true;
                Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "No longer forcing user-tracking features");
            }
        }



        private void CheckBox1_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox1.Checked)
            {
                My.MyProject.Forms.InstallerModule.Show();
            }
            else
            {
                My.MyProject.Forms.InstallerModule.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox2_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox2.Checked == true)
            {
                My.MyProject.Forms.BootManagerUXPort.Show();
            }
            else
            {
                My.MyProject.Forms.BootManagerUXPort.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox8_CheckedChanged(object sender, EventArgs e)
        {
            if (Label14.Text == "True")
            {
                if (CheckBox8.Checked == true)
                {
                    My.MyProject.Forms.UserInterface.Show();
                }
                else
                {
                    My.MyProject.Forms.UserInterface.Dispose();

                }
            }
            else
            {

            }

        }

        private void CheckBox7_CheckedChanged(object sender, EventArgs e)
        {
            if (Label14.Text == "True")
            {
                if (CheckBox7.Checked == true)
                {
                    My.MyProject.Forms.CoreForm.Show();
                }
                else
                {
                    My.MyProject.Forms.CoreForm.Dispose();



                }
            }
            else
            {

            }

        }

        private void CheckBox6_CheckedChanged(object sender, EventArgs e)
        {
            if (Label14.Text == "True")
            {
                if (CheckBox8.Checked == true)
                {
                    My.MyProject.Forms.UserPreferencesModule.Show();
                }
                else
                {
                    My.MyProject.Forms.UserPreferencesModule.Dispose();
                }
            }
            else
            {
            }

        }

        private void CheckBox5_CheckedChanged(object sender, EventArgs e)
        {

            if (CheckBox8.Checked == true)
            {
                My.MyProject.Forms.AutoPowerSwitcher.Show();
            }
            else
            {
                My.MyProject.Forms.AutoPowerSwitcher.Dispose();


            }


        }

        private void CheckBox12_CheckedChanged(object sender, EventArgs e)
        {

            switch (Interaction.MsgBox("Are you sure want to reset all the settings?", MsgBoxStyle.YesNo, "CryptedBytes Debugging Interface"))
            {
                case MsgBoxResult.Yes:
                    {
                        My.MySettingsProperty.Settings.Reset();
                        My.MySettingsProperty.Settings.Save();
                        break;
                    }

                case MsgBoxResult.No:
                    {
                        break;
                    }

            }


            CheckBox12.Checked = false;

        }

        private void CheckBox11_CheckedChanged(object sender, EventArgs e)
        {
            CheckBox11.Checked = false;

            throw new Exception("You have thrown an exception");

        }

        private void Statistics_Load(object sender, EventArgs e)
        {
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DeveloperForm is loaded into memory.");
            if (My.MySettingsProperty.Settings.DisableCoreForm == true)
            {
                CheckBox14.Checked = true;
            }
            else
            {
                CheckBox14.Checked = !true;

            }
            Panel1.Location = new Point(12, 291);
            if (My.MySettingsProperty.Settings.DisableLegacyServices == true)
            {
                CheckBox19.Checked = !true;
            }
            else
            {
                CheckBox19.Checked = !false;
            }

            if (My.MySettingsProperty.Settings.GroupPolicyForceAutoPilot == true)
            {
                CheckBox17.Checked = true;
            }
            if (My.MySettingsProperty.Settings.DisableSizeChangedEventChecks == true)
            {
                CheckBox18.Checked = true;

            }
            // Panel2.Location = New Point(4, 29)
            Debug.WriteLine(DateTime.Now.ToString("HH.mm.ss") + ": " + "DeveloperForm is intialized.");
        }

        private void CheckBox10_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox10.Checked == true)
            {
                CheckBox24.Checked = true;
            }
            else
            {
                CheckBox24.Checked = false;


            }
        }

        private void CheckBox4_CheckedChanged_1(object sender, EventArgs e)
        {
            if (CheckBox4.Checked)
            {
                CheckBox13.Checked = true;
            }
            else
            {
                CheckBox13.Checked = false;
            }
        }

        private void Timer1_Tick(object sender, EventArgs e)
        {
            if (Application.OpenForms.OfType<UserInterface>().Any())
            {
                CheckBox8.Checked = true;
            }
            else
            {
                CheckBox8.Checked = !true;

            }
            if (Application.OpenForms.OfType<CoreForm>().Any())
            {
                CheckBox7.Checked = true;
            }
            else
            {
                CheckBox7.Checked = !true;

            }
            if (Application.OpenForms.OfType<UserPreferencesModule>().Any())
            {
                CheckBox6.Checked = true;
            }
            else
            {
                CheckBox6.Checked = !true;

                if (Application.OpenForms.OfType<AutoPowerSwitcher>().Any())
                {
                    CheckBox6.Checked = true;
                }
                else
                {
                    CheckBox6.Checked = !true;

                }
            }
            Label14.Text = "True";
            GetOpenForms.Stop();
        }

        private void CheckBox9_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox9.Checked)
            {
                My.MyProject.Forms.BootManager.Show();
            }
            else
            {
                My.MyProject.Forms.BootManager.Dispose();
                GC.Collect();
            }
        }

        private void LinkLabel1_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Panel2.Dispose();
        }

        private void CheckBox3_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox3.Checked)
            {
                My.MyProject.Forms.Debugging.Show();
            }
            else
            {
                My.MyProject.Forms.Debugging.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox16_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox16.Checked)
            {
                My.MyProject.Forms.MemoryLeakTestingLab.Show();
            }
            else
            {
                My.MyProject.Forms.MemoryLeakTestingLab.Dispose();
                GC.Collect();
            }
        }

        private void CheckBox14_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox14.Checked)
            {
                My.MySettingsProperty.Settings.DisableCoreForm = true;
            }
            else
            {
                My.MySettingsProperty.Settings.DisableCoreForm = false;

            }
            My.MySettingsProperty.Settings.Save();

        }

        private void CheckBox17_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox17.Checked)
            {
                My.MySettingsProperty.Settings.GroupPolicyForceAutoPilot = true;
                My.MyProject.Forms.UserPreferencesModule.Userdetect.Checked = true;
                My.MyProject.Forms.UserPreferencesModule.Userdetect.Enabled = false;
                My.MyProject.Forms.UserPreferencesModule.UserNOdetect.Checked = false;
                My.MyProject.Forms.UserPreferencesModule.UserNOdetect.Enabled = false;
            }
            else
            {
                My.MyProject.Forms.UserPreferencesModule.Userdetect.Checked = false;
                My.MyProject.Forms.UserPreferencesModule.Userdetect.Enabled = true;
                My.MyProject.Forms.UserPreferencesModule.UserNOdetect.Checked = false;
                My.MyProject.Forms.UserPreferencesModule.UserNOdetect.Enabled = true;
                My.MySettingsProperty.Settings.GroupPolicyForceAutoPilot = false;

            }

            My.MySettingsProperty.Settings.Save();


        }

        private void LinkLabel2_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
        {
            Panel1.Dispose();
        }

        private void CheckBox19_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox19.Checked == true)
            {
                My.MySettingsProperty.Settings.DisableLegacyServices = false;
                Interaction.MsgBox("My.Settings.DisableLegacyServices = False", MsgBoxStyle.Exclamation, "Boolean Value Set");
            }
            else
            {
                My.MySettingsProperty.Settings.DisableLegacyServices = true;
                Interaction.MsgBox(" My.Settings.DisableLegacyServices = True", MsgBoxStyle.Exclamation, "Boolean Value Set");
            }

        }

        private void RadioButton2_CheckedChanged(object sender, EventArgs e)
        {

        }

        private void CheckBox18_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox18.Checked == true)
            {
                My.MySettingsProperty.Settings.DisableSizeChangedEventChecks = true;
            }
            else
            {
                My.MySettingsProperty.Settings.DisableSizeChangedEventChecks = false;
            }

        }

        private void Button1_Click(object sender, EventArgs e)
        {


        }

        private void CheckBox20_CheckedChanged(object sender, EventArgs e)
        {
            Environment.FailFast(1.ToString());
        }

        private void CheckBox21_CheckedChanged(object sender, EventArgs e)
        {

            if (CheckBox21.Checked == true)
            {
                GC.Collect();
                Interaction.MsgBox("Succesful.");
                CheckBox21.Checked = false;
            }
            else
            {

            }

        }

        private void CheckBox22_CheckedChanged(object sender, EventArgs e)
        {
            if (CheckBox22.Checked == true)
            {
                CoreForm.errorLoggingText = DateTime.Now.ToString("dd.MM.yyyy HH:mm.ss") + ": " + "CRITICAL: User thrown an exception";
                CoreForm.errorLoggingSystem();
                CheckBox22.Checked = false;

            }
        }

        private void CheckBox23_CheckedChanged(object sender, EventArgs e)
        {
            Interaction.MsgBox("Form1's exact size is: " + My.MyProject.Forms.UserInterface.FormHeightLabel.Text + " , " + My.MyProject.Forms.UserInterface.FormWidthLabel.Text + " and is copied to your clipboard.");
            My.MyProject.Computer.Clipboard.SetText(My.MyProject.Forms.UserInterface.FormHeightLabel.Text + " , " + My.MyProject.Forms.UserInterface.FormWidthLabel.Text);
            CheckBox23.Checked = false;
        }
    }
}